src/client.js

```
@@ -131,6 +149,20 @@ let pushClient = {
     navigator.serviceWorker.removeEventListener('message', messageHandler);
   },
 
+  async getSubscription() {
+    if (!this.supported()) {
+      return;
+    }
+
+    let registration = await getRegistration();
+
+    if (!registration) {
+      return;
```

Same as above - should we return an error?